### PR TITLE
Implementation of request module

### PR DESCRIPTION
**Pull request by shdshdw**
2: As a housing corporation agent I want to be able to react on denials or approvals of sell/buy requests
so that I can adjust my strategy based on other stakeholders' responses. 

Note that this PR also includes some minor bugfixes.
Also note that the request strategy for denied buy land requests does not get triggered.

### CODE REVIEW

File: src/nl/codefox/contextvh/tygronGoals.mod2g [hunk 1]
```diff
@@ -4,6 +4,8 @@ order = linearall.
 module tygronGoals {
 	%_______________________________Adopt goals based on the environment_______________________________%
 	
+	if true then adopt(permaGoal).
```
Comment (MHMCD) [inline src/nl/codefox/contextvh/tygronGoals.mod2g]: Use exit=never instead of adding a goal which can not be accomplished.

**Comment by MHMCD** [comment]
looks good now, can be merged

**Comment by JoshCode** [comment]
Changes look good. My only, yet important remark:

This PR doesn't ONLY implement the request module but also does a lot of little things, some trivial and only affect the code style, some not so trivial that affect the bot behaviour outside of the PR specifications, like the upgrade behaviour. **Don't do this next time.**

**Comment by MatthijsJWH** [comment]
Changes and implemantation look good. I see no tests however. Have you looked into automatic testing, or have you concluded it to be nearly impossible to do automatic testing?

**Comment by shdshdw** [comment]
@MatthijsJWH, Automatic testing is going to be hard, because requests rely on interaction with other stakeholders. As for these little bugfixes and changes you are right @JoshCode, I should have included these changes in another branch, but I was a little lazy because some of the other code in this PR is dependent.

**Comment by MatthijsJWH** [comment]
Thanks for the clarification on testing, can now be merged in my opinion

**Pull request merged**